packages/raja/package.py

```diff
@@ -9,9 +9,93 @@
 import re
 
 from spack.package import *
-from spack.pkg.builtin.camp import hip_for_radiuss_projects
-from spack.pkg.builtin.camp import cuda_for_radiuss_projects
-from spack.pkg.builtin.camp import blt_link_helpers
+
+
+def spec_uses_toolchain(spec):
+    gcc_toolchain_regex = re.compile(".*gcc-toolchain.*")
+    using_toolchain = list(filter(gcc_toolchain_regex.match, spec.compiler_flags["cxxflags"]))
+    return using_toolchain
+
+def spec_uses_gccname(spec):
+    gcc_name_regex = re.compile(".*gcc-name.*")
+    using_gcc_name = list(filter(gcc_name_regex.match, spec.compiler_flags["cxxflags"]))
+    return using_gcc_name
+
+def hip_for_radiuss_projects(options, spec, spec_compiler):
+    # Here is what is typically needed for radiuss projects when building with rocm
+    hip_root = spec["hip"].prefix
+    rocm_root = hip_root + "/.."
+    options.append(cmake_cache_path("HIP_ROOT_DIR", hip_root))
+    options.append(cmake_cache_path("ROCM_ROOT_DIR", rocm_root))
+
+    hip_repair_cache(options, spec)
+
+    archs = spec.variants["amdgpu_target"].value
+    if archs != "none":
+        arch_str = ",".join(archs)
+        options.append(
+            cmake_cache_string("HIP_HIPCC_FLAGS", "--amdgpu-target={0}".format(arch_str))
+        )
+        options.append(
+            cmake_cache_string("CMAKE_HIP_ARCHITECTURES", arch_str)
+        )
+
+    # adrienbernede-22-11:
+    #   Specific to Umpire, attempt port to RAJA and CHAI
+    hip_link_flags = ""
+    if "%gcc" in spec:
+        gcc_bin = os.path.dirname(spec_compiler.cxx)
+        gcc_prefix = join_path(gcc_bin, "..")
+        options.append(cmake_cache_string("HIP_CLANG_FLAGS", "--gcc-toolchain={0}".format(gcc_prefix)))
+        options.append(cmake_cache_string("CMAKE_EXE_LINKER_FLAGS", hip_link_flags + " -Wl,-rpath {}/lib64".format(gcc_prefix)))
+    else:
+        options.append(cmake_cache_string("CMAKE_EXE_LINKER_FLAGS", "-Wl,-rpath={0}/llvm/lib/".format(rocm_root)))
+
+def cuda_for_radiuss_projects(options, spec):
+    # Here is what is typically needed for radiuss projects when building with cuda
+
+    cuda_flags = []
+    if not spec.satisfies("cuda_arch=none"):
+        cuda_arch = spec.variants["cuda_arch"].value
+        cuda_flags.append("-arch sm_{0}".format(cuda_arch[0]))
+        options.append(
```

Comment:
If CUDAPackage isn't doing this already, this looks ripe being included into that class.

Reply:
newer CMakes do not need this. You just do this:

```
set(CMAKE_CUDA_ARCHITECTURES "70" CACHE STRING "")
```

Reply:
True, if radiuss does not need to maintain compatibility with older cmake some of this can be cleaned up.

Reply:
@aaroncblack I believe CMake 3.14.5, which is the default on most LC systems requires some of this stuff to work. That's why it's still in the RAJA Spack package.